tests/test_schemas.py

```diff
@@ -1,294 +1,645 @@
+            json_validation_schema = dmjs.get_json_validation_schema(source_node=source_node, schema_name=schema_name)
+
+            # Check Keys in Schema
+            expected_jvs_keys = ['$schema', '$id', 'title', 'type', 'properties', 'required', 'allOf']
+            actual_jvs_keys = list( json_validation_schema.keys())
+            assert expected_jvs_keys == actual_jvs_keys
+
+            # Check title
+            assert schema_name == json_validation_schema['title']
+
+            # Check contents of validation schema
+            assert 'Diagnosis' in json_validation_schema['properties']
+            assert 'Cancer' in json_validation_schema['properties']['Diagnosis']['enum']
+        except:
+            # Should only fail if no source node is provided.
+            assert source_node == ''
+
+class TestDataModelJsonLd:
+    @pytest.mark.parametrize("data_model", list(DATA_MODEL_DICT.keys()), ids=list(DATA_MODEL_DICT.values()))
+    def test_init(self, helpers, data_model):
+        # Get Graph
+        graph_data_model = generate_graph_data_model(helpers, data_model_name=data_model)
+
+        # Instantiate DataModelJsonLD
+        data_model_jsonld = DataModelJsonLD(Graph=graph_data_model)
+
+        # Test that __init__ is being set up properly 
+        assert type(data_model_jsonld.graph) == nx.MultiDiGraph
+        assert type(data_model_jsonld.rel_dict) == dict
+        assert 'required' in data_model_jsonld.rel_dict
+        assert type(data_model_jsonld.DME) == DataModelGraphExplorer
+        assert data_model_jsonld.output_path == ''
+
+    @pytest.mark.parametrize("data_model", list(DATA_MODEL_DICT.keys()), ids=list(DATA_MODEL_DICT.values()))
+    def test_base_jsonld_template(self, helpers, data_model):
+        # Get Graph
+        graph_data_model = generate_graph_data_model(helpers, data_model_name=data_model)
+
+        # Instantiate DataModelJsonLD
+        data_model_jsonld = DataModelJsonLD(Graph=graph_data_model)
+
+        # Get base jsonld template
+        base_template = data_model_jsonld.base_jsonld_template()
+
+        # Test base template is constructed as expected
+        assert '@context' in base_template
+        assert '@graph' in base_template
+        assert '@id' in base_template
+
+    @pytest.mark.parametrize("data_model", list(DATA_MODEL_DICT.keys()), ids=list(DATA_MODEL_DICT.values()))
+    @pytest.mark.parametrize("template_type", ['property', 'class'], ids=['property', 'class'])
+    @pytest.mark.parametrize("node", ['', 'Patient'], ids=['no node', 'Patient'])
+    def test_create_object(self, helpers, data_model, template_type, node):
+        # Get Graph
+        graph_data_model = generate_graph_data_model(helpers, data_model_name=data_model)
+
+        # Instantiate DataModelJsonLD
+        data_model_jsonld = DataModelJsonLD(Graph=graph_data_model)
+
+        # Get empty template
+        if template_type == 'property':
+            template = data_model_jsonld.property_template()
```

Comment:
Is checking all the expected keys got returned sufficient? Or should we also check the values of the key to make sure that they are expected too?

Migrated from: https://github.com/Sage-Bionetworks/schematic/pull/1313